### ISSUE TITLE
Trying to build for Debian ARM64, asks for 'Microsoft.WindowsDesktop.App'

### ISSUE THREAD

**Issue by drunkinlove** [lay295/TwitchDownloader#865]
### Checklist

- [X] I have checked the [issue page](https://github.com/lay295/TwitchDownloader/issues?q=is%3Aissue) for duplicates
- [X] I am running the latest version ([download here](https://github.com/lay295/TwitchDownloader/releases/latest))

### Edition

Command Line Interface

### Describe your issue here

Hi!
I'm trying to use this on Debian 11 in a VM (UTM, latest) on an M2 Macbook (probably an obscure setup). 
1) The [TwitchDownloaderCLI-1.53.4-LinuxArm64.zip](https://github.com/lay295/TwitchDownloader/releases/download/1.53.4/TwitchDownloaderCLI-1.53.4-LinuxArm64.zip) binary throws a `Segmentation fault`
2) When I try building from source using the `LinuxArm64` profile, I get a binary that prints the following on launch:

```
You must install or update .NET to run this application.

App: /home/nx/TwitchDownloader/TwitchDownloaderWPF/bin/Debug/net6.0-windows/publish/TwitchDownloaderWPF
Architecture: arm64
Framework: 'Microsoft.WindowsDesktop.App', version '6.0.0' (arm64)
.NET location: /home/nx/.dotnet

No frameworks were found.

Learn about framework resolution:
https://aka.ms/dotnet/app-launch-failed

To install missing framework, download:
https://aka.ms/dotnet-core-applaunch?framework=Microsoft.WindowsDesktop.App&framework_version=6.0.0&arch=arm64&rid=debian.11-arm64
```

The compiled binary is located at `TwitchDownloader/TwitchDownloaderWPF/bin/Debug/net6.0-windows/publish/TwitchDownloaderWPF`. 

Could you help me build this or make the precompiled binary work?

**Comment by ScrubN** [comment]
I do not own an arm machine to test the arm64 binary, that was added by a contributor, however if you could provide more details about the segfault and possibly even a .dmp file of the crash, I may be able to diagnose the issue.

As for TwitchDownloaderWPF, that is a Windows binary. I would suggest trying TwitchDownloaderCLI-1.53.4-LinuxArm.zip before trying to run the WPF application with wine.

**Comment by drunkinlove** [comment]
With the LinuxArm version, I get the following:
`-bash: ./TwitchDownloaderCLI: cannot execute binary file: Exec format error`
(which is weird, I thought it should run fine in compatibility mode?)

Here's the core dump that was generated after I set `ulimit -c unlimited`:
https://drive.google.com/file/d/18vULdkkZewaEOpg0rgFWrdmC6pIBplfh/view?usp=share_link

**Comment by ScrubN** [comment]
> Here's the core dump that was generated after I set `ulimit -c unlimited`:
> https://drive.google.com/file/d/18vULdkkZewaEOpg0rgFWrdmC6pIBplfh/view?usp=share_link

Unfortunately the dump lacks a heap snapshot, so the only information it can provide is the OS architecture and loaded system libraries.

**Comment by superbonaci** [comment]
I've downloaded [Debian 11 (Xfce) ARM64](https://mac.getutm.app/gallery/debian-11-xfce) to run inside UTM, my ARM machine is Apple M1. It runs perfectly fine.

ARM unlike x86 or x86_64 is different for each CPU, so maybe yours is incompatible.
If you were so kind to tell us which CPU is yours?

My CPU reported by hyfetch is:
```
CPU: 0x00 Architecture: aarch64 (4)
```
Maybe the cause is UTM. I've downloaded mine from Apple Store.

**Comment by drunkinlove** [comment]
@superbonaci hi! I'm on M2. hyfetch says the same (and I got a nice flag):
```
OS: Debian GNU/Linux 11 (bullseye) aarch64
CPU: 0x00 Architecture: aarch64 (4)
```
I installed UTM through the .dmg [on their github](https://github.com/utmapp/UTM/releases/tag/v4.4.3).

This is kind of irrelevant, but the x64 version of TwitchDownloader [worked for me today all of a sudden](https://github.com/lay295/TwitchDownloader/issues/865#issuecomment-1772210835), I could download a VOD. 
(This is in docker by the way, not UTM. Also ignore that remark about Apple backend vs qemu as qemu also uses the former as I now understand).

I still get a segfault in ARM64 Debian (UTM) though.

**Comment by superbonaci** [comment]
@drunkinlove I'm not sure what you are doing, are you on MacOS 14.0?
Could you try with the same debian iso as me [Debian 11 (Xfce) ARM64](https://mac.getutm.app/gallery/debian-11-xfce)?
Are you using this version https://github.com/lay295/TwitchDownloader/releases/download/1.53.4/TwitchDownloaderCLI-1.53.4-LinuxArm64.zip?

I don't think the M1 vs the M2 can be the responsible...

Does you native binary work on MacOS https://github.com/lay295/TwitchDownloader/releases/download/1.53.4/TwitchDownloaderCLI-1.53.4-MacOS-x64.zip?

**Comment by drunkinlove** [comment]
@superbonaci I'm on Sonoma, yes, and I'm using the binary you linked. My Debian image is the ARM64 version from the official website. I'll try the one hosted by UTM if I have time...
I'd rather not use unsigned apps on this Mac besides what's already installed, which is why I run stuff in VMs.

**Comment by superbonaci** [comment]
@drunkinlove yes find some time for it please. Maybe the ARM64 from Debian website is for another cpu.

**Comment by drunkinlove** [comment]
@superbonaci I tried this and it worked! 🤔
This is the first app I've had problems with on the official Debian image; ffmpeg, python, mp4box and others ran fine. Although I'm pretty sure I had to build everything else from source

**Comment by superbonaci** [comment]
@drunkinlove there are 3 ARM versions of debian for ARM:
https://cdimage.debian.org/debian-cd/current/

- arm64
- armel
- armhf

Maybe you should ask in [Debian Forums](https://forums.debian.net/) or follow the [UTM guide for Debian](https://docs.getutm.app/guides/debian/). There must be some bug somewhere.

**Comment by drunkinlove** [comment]
The armel/armhf are for 32-bit architectures:

> Debian fully supports three ports to different flavours of little-endian ARM hardware:
> 
> The [ARM EABI](https://wiki.debian.org/ArmEabiPort) (armel) port targets a range of older 32-bit ARM devices, particularly those used in NAS hardware and a variety of *plug computers.
> The newer [ARM hard-float](https://wiki.debian.org/ArmHardFloatPort) (armhf) port supports newer, more powerful 32-bit devices using version 7 of the ARM architecture specification.
> The [64-bit ARM](https://wiki.debian.org/Arm64Port) (arm64) port supports the latest 64-bit ARM-powered devices.

My problem might be related to the [SSL certificate chain](https://github.com/lay295/TwitchDownloader/issues/865#issuecomment-1772210835) and probably lies in the installation method. I'll report if I learn something new...

**Comment by superbonaci** [comment]
@drunkinlove hope you find the bug. I'd love to help you but I'm already into lots of bugs that need fix.
